google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/gcs/GcsBucketContentEditorPanel.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.gcs;
+
+import com.google.cloud.storage.Blob;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Iterators;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Vector;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+import javax.swing.JPanel;
+import javax.swing.JTable;
+import javax.swing.table.DefaultTableModel;
+import org.jetbrains.annotations.NotNull;
+
+/** Defines the Google Cloud Storage bucket content browsing UI panel. */
+// TODO(eshaul) work in progress
+final class GcsBucketContentEditorPanel {
+
+  private JPanel bucketContentEditorPanel;
+  private JPanel bucketContentToolbarPanel;
+  private JPanel breadCrumbsPanel;
+  private JTable bucketContentTable;
+
+  private static final List<String> TABLE_COL_HEADER =
+      Arrays.asList("Name", "Size", "Type", "Last Modified");
+
+  GcsBucketContentEditorPanel(@NotNull GcsBucketVirtualFile bucketVirtualFile) {
+    Iterable<Blob> blobs = bucketVirtualFile.getBucket().list().iterateAll();
```

Comment:
Generally, it's discouraged to have constructors do any real work because it "removes seams needed for testing, forcing subclasses/mocks to inherit unwanted behavior. Too much work in the constructor prevents instantiation or altering collaborators in the test." - go/cl-constructors

What about having an instance field for this `GcsBucketVirtualFile` and expose a method to fill the table with data?

Reply:
the table model needs to be set on initialization of the panel. I could perhaps move the setting of the model to the `.form` itself but that's not really helping your concern. Maybe I didn't understand your suggestion?

Reply:
This class is instantiated by `GcsBucketContentEditor`, so you control the lifecycle of this object. I'm suggesting to move out the API call with the model initialization to an instance method and call the method from `GcsBucketContentEditor` when you want (which in this case, might be right after you initialize it).